datafusion/core/src/physical_plan/union.rs

```diff
@@ -194,6 +258,73 @@ impl ExecutionPlan for UnionExec {
     }
 }
 
+/// CombinedRecordBatchStream can be used to combine a Vec of SendableRecordBatchStreams into one
```

Comment:
I feel there was already a piece of code that does this -- maybe @tustvold  can remind me 🤔